cabal-testsuite/PackageTests/NewUpdate/UpdateIndexState/update-index-state.test.hs

```diff
@@ -1,6 +1,8 @@
 import Test.Cabal.Prelude
 
 main = cabalTest $ withRemoteRepo "repo" $ do
+  skipIf "osx" =<< isOSX
```

Comment:
https://github.com/haskell/cabal/blob/6df9ab4f31061d9c87b17d61f58876752efb419a/cabal-testsuite/README.md#expect-tests

> If you can't get the output of a test to be deterministic, no problem: just exclude it from recording and do a manual test on the output for the string you're looking for. Try to be deterministic, but sometimes it's not (easily) possible.

Since the warning is not part of the test expectations, instead of duplicating the test I think we can just add `recordMode DoNotRecord`, `assertOutputContains "The index-state is set to 2022-01-28T02:36:41Z"`, and maybe `assertOutputDoesNotContain "revert"` to the first update call.

Reply:
@fgaz I'll see what I can come up with, bit more than I bargained for :p Does that affect all lines in the test, or just for the first of the three tests?

Reply:
Just the first